R/utils.R

```diff
@@ -259,85 +259,560 @@ rowcol_to_dim <- function(row, col) {
+  # from_row / from_col = 0 only acceptable in certain cases.
+  if (!all(length(scol) == 1, length(srow) == 1)) {
+    stop("Internal error. At this point scol and srow should have length 1.")
+  }
+  if (!x_present && (identical(scol, -1L) || identical(srow, -1L))) {
+    stop("`from_row/col` = 0 only makes sense with `x` present")
+  }
+
+
+  # if `!x` return early
+  if (!x_present) {
+    row_span <- srow + rows_arg %||% 1L
+    col_span <- scol + cols_arg %||% 1L
+    if (identical(row_span, 0L)) {
+      stop("Providing `rows = 0` without an object with dimensions is not supported", "Use `rows = 1`.")
+    }
+    if (identical(col_span, 0L)) {
+      stop("Providing `cols = 0` without an object with dimensions is not supported", "Use `cols = 1`.")
+    }
+
+    if (length(row_span) == 1 && length(col_span) == 1) {
+      # A1
+      row_start <- row_span
+      col_start <- col_span
+      dims <- rowcol_to_dim(row_start, col_start)
+    } else {
+      # A1:B2
+      dims <- rowcol_to_dims(row_span, col_span)
+    }
+    return(dims)
+  }
 
+  # Fix an error I created
+  # if (frow_null)  {
+  #   srow <- srow + 1L
+  # }
+  # if (fcol_null) {
+  #   scol <- scol + 1L
+  # }
+
+
+
+  # After this point, we only cover the case for `x`
+  rows_arg
+  cols_arg
+  if (cnam_null && x_has_named_dims) {
+    if (identical(rows_arg, 0L)) {
+      # message("Use `col_names = TRUE` explicitly to select `x + its column names`",
+      #        "\nBy default, when `x` is specified, `rows` and `cols` are only related to the content of `x`.")
+      # args$col_names <- FALSE
+    } else if (is.null(rows_arg)) {
+      # message(
+      #   "Use `col_names = TRUE` explicitly to select `x + its column names`",
+      #   "\nBy default, when `x` is specified, `rows` and `cols` are only related to the content of `x`."
+      # )
+      args$col_names <- FALSE
+    } else {
+      # message(
+      #   "Use `col_names = TRUE` explicitly to select `x + its column names`",
+      #   "\nBy default, when `x` is specified, `rows` and `cols` are only related to the content of `x`."
+      # )
+      args$col_names <- FALSE
+    }
   }
 
-  rows <- rows + srow
-  cols <- cols + scol
+  col_names <- args$col_names %||% x_has_named_dims
+
+  if (x_present && !col_names && x_has_named_dims && !cnam_null) {
+    if (x_has_colnames) {
+      warning("`x` has column names. Yet, you are asking for `col_names = FALSE`.",
+        "\n ",
+        "\n Consider supplying `x = unname(`input`)`, or use `wb_dims()` without `x` to ensure no errors with `col_names = FALSE`",
+        call. = FALSE
+      )
+    } else {
+      # message("`x` doesn't have col names. assuming there is no name. Supply `col_names = TRUE` only to select rows + column name.")
+    }
+  }
 
-  if (length(rows) == 1 && length(cols) == 1) {
-    # A1
-    dims <- rowcol_to_dim(rows, cols)
+  if (!cnam_null && !x_has_named_dims) {
+    stop("Supplying `col_names` when `x` is a vector is not supported.")
```

Comment:
Unfortunately having something like this might be required. I do not recall from the top of my head, but there is a case where we write characters with column names. But it might be with transposed vectors, which are matrices and not vectors anymore. Anyhow it should not be an error, maybe a message that col_names with a vector was selected.

Reply:
I would be careful before loosening up, (I want to have tests for use cases as much as possible !) But I wouldn't mind, when a use case arises